### PR TITLE
ROC-7696 Add paymentDate and paymentMethod to defence response pdf

### PR DESCRIPTION
**Pull request by justin-r-oliver**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7696

### Change description ###
1. Added paymentDate and paymentMethod to defence response PDF for states paid 
2. Formatted document template to improve readability e.g. comments, indenting

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by andrewwa-kainos** [comment]
Please note the branch should have been under `features/`
i.e.
`features/ROC-7696_Add-paymentDate-and-paymentMethod-to-defence-response-pdf`
(though something less wordy wouldn't be amiss)

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 1]
```diff
@@ -1236,7 +1257,7 @@ <h2 class="heading-medium">
           <strong>Role:</strong> {{ signerRole }}
           {% endif %}
 
-          <br/>{{ defenceSubmittedOn }}
+          <br/><span id="defenceSubmittedOn">{{ defenceSubmittedOn }}</span>
```
Comment (andrewwa-kainos) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html:1260]: An HTML document should only have unique `id`s. (Repeated on line [928](https://github.com/hmcts/cmc-claim-store/pull/1612/files#diff-1c9530b46501cb30d2f915bea3c5dcafR928).) Are either of them necessary?

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 2]
```diff
@@ -1168,7 +1189,7 @@ <h2 class="heading-medium">
       </td>
       <td>
         <div>
-          {{ hearingContent.witness.selfWitness | capitalize }}
+          {{ hearingContent.witness.selfWitness.toString | capitalize }}
```
Comment (andrewwa-kainos) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html:1192]: Is this change necessary?

Comment (justin-r-oliver) [reply src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: Removed until next PR

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html (hunk 1 above)
Comment (justin-r-oliver) [reply src/main/resources/citizen/templates/document/defendantResponseReceipt.html]: Removed until next PR

File: src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/FullDefenceResponseContentProvider.java [hunk 3]
```diff
@@ -53,8 +52,10 @@ public FullDefenceResponseContentProvider(HearingContentProvider hearingContentP
         }
 
         fullDefenceResponse.getPaymentDeclaration()
-            .map(this::createContentFor)
-            .ifPresent(paymentContent -> content.put("paymentDeclaration", paymentContent));
+            .ifPresent(paymentDeclaration -> {
+                content.put("paymentDate", formatDate(paymentDeclaration.getPaidDate()));
+                content.put("paymentMethod", paymentDeclaration.getExplanation());
```
Comment (karthikkannankainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/documents/content/FullDefenceResponseContentProvider.java:57]: is payment explanation missed out? paymentDeclaration.getExplanation()

File: src/main/resources/citizen/templates/document/defendantResponseReceipt.html [hunk 4]
```diff
@@ -401,13 +402,29 @@
...
+            </tr>
+          {% endif %}
+          {% if paymentMethod is defined and paymentMethod is not empty %}
+            <tr>
+              <td>How they paid the amount</td>
+              <td>
+                {{ paymentMethod | trim }}
+              </td>
+            </tr>
+        {% endif %}
```
Comment (karthikkannankainos) [inline src/main/resources/citizen/templates/document/defendantResponseReceipt.html:427]: Format pls

**Comment by justin-r-oliver** [comment]
Replaced by https://github.com/hmcts/cmc-claim-store/pull/1633
